modules/gitops-release-notes-1-17-0.adoc

```diff
@@ -0,0 +1,170 @@
+=== Upgrade to Argo CD 3.0
+
+This release introduces an upgrade to the Argo CD component, transitioning from v2.14 to v3.0. This is a minor upgrade that introduces a list of breaking changes in Argo CD v3.0, with remediation steps provided in the following section. For more information, see link:https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.14-3.0/[Upgrading Argo CD from v2.0 to v3.0].
+
+[id="rbac-behavior-changes-with-argo-cd-dex-sso_{context}"]
+=== RBAC behavior changes with Argo CD 3.0 and Dex SSO
+
+* Argo CD 3.0 introduces a change in RBAC handling when using Dex for single sign-on (SSO). The RBAC system uses the `federated_claims.user_id` field instead of the previously encoded sub claim. This change affects how user identities are evaluated and matched in RBAC policies.
++
+To address this, the Argo CD Operator documentation has been updated with:
+
+** Detailed migration guidance for transitioning from sub to federated_claims.user_id
+** Step-by-step remediation procedures
+** Practical RBAC policy examples
+** Best practices for managing RBAC with the new hands-off approach
++
+Users must review and update their RBAC policies to ensure compatibility with Argo CD 3.0 when using the {gitops-title} Operator. For more information, see link:https://argocd-operator.readthedocs.io/en/latest/upgrading/#rbac-with-dex-sso-authentication[Configuring RBAC with Dex SSO Authentication].
++
+--
+link:https://issues.redhat.com/browse/GITOPS-6887[GITOPS-6887]
+--
+
+[id="added-guidance-for-logs-rbac-enforcement-in-argocd_{context}"]
+=== Added guidance for logs RBAC enforcement in Argo CD 3.0
+
+* Before this update, the Argo CD Operator upstream documentation did not explain the changes to logs RBAC introduced in Argo CD 3.0. In contrast to Argo CD 2.x, where users with access to Applications could also view logs, Argo CD 3.0 treats logs as a distinct RBAC resource requiring explicit permissions. With this update, the Argo CD Operator upstream documentation includes:
+
+** A detailed explanation of logs-specific RBAC enforcement
+** Migration guides and remediation steps
+** Examples for updating policies to include logs permissions
+** Best practices for RBAC under the new hands-off management model
++
+--
+link:https://issues.redhat.com/browse/GITOPS-6886[GITOPS-6886]
+--
+
+[id="gitops-1-17-0-fine-grained-rbac-inheritance-enabled-by-default_{context}"]
+=== Fine-grained RBAC inheritance enabled by default
+
+* Starting with Argo CD v3.0, the `update` and `delete` actions apply only to the application resource. To perform these actions on managed resources, you must configure policies to allow `update/\*` or `delete/\*`. To maintain the Argo CD v2.0 behavior, the `server.rbac.disableApplicationFineGrainedRBACInheritance` parameter is set to `false` by default in the `argocd-cm` config map. You can enable the new fine-grained RBAC functionality by setting the following configuration in the Argo CD custom resource:
```

Comment:
```suggestion
* Starting with Argo CD v3.0,  fine grained RBAC inheritance of Applications have been disabled making `update` and `delete` actions apply only to the application resource. To perform these actions on managed resources, you must configure new policies. To maintain the Argo CD v2.0 behavior, the `server.rbac.disableApplicationFineGrainedRBACInheritance` parameter is set to `false` by default in the `argocd-cm` config map to preserve the old RBAC behaviour. You can enable the new fine-grained RBAC functionality by setting the following configuration in the Argo CD custom resource:
```

Reply:
cc: @anandf

Reply:
@anandf should we mention upstream documentation as well?
https://argocd-operator.readthedocs.io/en/latest/reference/argocd/#fine-grained-rbac-for-application-update-and-delete-sub-resources-v30